### PR TITLE
Authorize installation on PHP8

### PR DESCRIPTION
**Pull request by Th3Mouk**

### CODE REVIEW

**Comment by Th3Mouk** [comment]
Hi @webmozart ! I know you are probably busy but this one can be quickly merged IMO

**Comment by webmozart** [comment]
@Th3Mouk thanks! Could you take care of the failing builds?

**Comment by Th3Mouk** [comment]
@webmozart I took the configuration of `assert` repo but a composer issue makes the build fail randomly I didn't found how to resolve it. If it's a problem with composer mirrors maybe the problem will be gone tomorrow.

File: .travis.yml [hunk 1]
```diff
@@ -1,38 +1,38 @@
...
+    - php: 7.2
+    - php: 7.3
+      env:
+        - COVERAGE=yes
+        - ANALYSIS=yes
+    - php: 7.4
+    - php: hhvm-3.30
+    - php: nightly
   allow_failures:
-    - php: hhvm
```
Comment (samdark) [inline .travis.yml:26]: This should be added back. Tests will silently fail HHVM as they did previously. See this job from master branch: https://travis-ci.org/github/webmozart/glob/jobs/152415050

File: .travis.yml [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
-      env: COVERAGE=yes
-    - php: 7.0
-      env: COMPOSER_FLAGS='--prefer-lowest --prefer-stable'
+    - php: 7.1
+    - php: 7.2
+    - php: 7.3
+      env:
+        - COVERAGE=yes
+        - ANALYSIS=yes
+    - php: 7.4
```
Comment (samdark) [inline .travis.yml:23]: I think tests should be performed on PHP 8 as well.

Comment (Th3Mouk) [reply .travis.yml:23]: nightly is php 8 but it exists probably a better tag now

Comment (samdark) [reply .travis.yml:23]: As an alternative, tests could be moved to GitHub actions. See https://github.com/yiisoft/strings/tree/master/.github/workflows as an example. There, you'll have PHP 8 beta. But, that's better to be done separately.

Comment (Th3Mouk) [reply .travis.yml:23]: I saw many projects doing this, this month. We are on GA too in my projects. But I think we must first resolve PHP8 installation/CI and then refacto the CI process

Comment (mnapoli) [reply .travis.yml:23]: We can now use `8.0` in Travis.

Comment (samdark) [reply .travis.yml:23]: Travis isn't giving enough credits to OpenSource anymore so a switch to GitHub actions is inevitable.

Comment (mnapoli) [reply .travis.yml:23]: This could be done separately, it's not related to this PR.

Comment (samdark) [reply .travis.yml:23]: Then it's ready to merge except https://github.com/webmozart/glob/pull/19/files#r511509629 should be applied.

Comment (Th3Mouk) [reply .travis.yml:23]: Ok then I will make the modif

**Comment by martinssipenko** [comment]
Should be closed in favor of #24, @Ocramius will you please close it?

**Pull request closed without merging**